plugins/basic/nexus-repository-docker/src/main/java/org/sonatype/nexus/repository/docker/DockerFacet.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Sonatype Nexus (TM) Open Source Version
+ * Copyright (c) 2008-present Sonatype, Inc.
+ * All rights reserved. Includes the third-party code listed at http://links.sonatype.com/products/nexus/oss/attributions.
+ *
+ * This program and the accompanying materials are made available under the terms of the Eclipse Public License Version 1.0,
+ * which accompanies this distribution and is available at http://www.eclipse.org/legal/epl-v10.html.
+ *
+ * Sonatype Nexus (TM) Professional Version is available from Sonatype, Inc. "Sonatype" and "Sonatype Nexus" are trademarks
+ * of Sonatype, Inc. Apache Maven is a trademark of the Apache Software Foundation. M2eclipse is a trademark of the
+ * Eclipse Foundation. All other trademarks are the property of their respective owners.
+ */
+package org.sonatype.nexus.repository.docker;
+
+import java.io.IOException;
+import java.util.Map;
+
+import org.sonatype.nexus.repository.Facet;
+import org.sonatype.nexus.repository.view.Payload;
+
+/**
+ * Provides persistence for Docker V1 protocol.
+ *
+ * @since 3.0
+ */
+@Facet.Exposed
+public interface DockerFacet
+    extends Facet
+{
+  /**
+   * Retrieve images index.
+   *
+   * @param name docker repository name
+   * @return image index JSON
+   */
+  Payload getImagesIndex(String name) throws IOException;
```

Comment:
Underlying imple uses `Content` but iface declares `Payload`, which is OK.

Am unsure how proxy will be built, but if you will want to use `ProxyFacetSupport` (I believe that will be the case), you will need `Content` for it. Luckily, as I said, imple uses it, so just change return type ;)